File: 2670. Find the Distinct Difference Array/Solution.py
class Solution:
    def distinctDifferenceArray(self, nums: List[int]) -> List[int]:
        n, p, s = len(nums), 0, 0
        prefix, suffix, res = [0 for i in range(51)], [0 for i in range(51)], [0 for i in range(n)]
        for i in range(n):
            suffix[nums[i]] += 1
            if suffix[nums[i]] == 1: s += 1
        for i in range(n):
            suffix[nums[i]] -= 1
            prefix[nums[i]] += 1
            if suffix[nums[i]] == 0: s -= 1
            if prefix[nums[i]] == 1: p += 1
            res[i] = p - s
        return res 

File: 2670. Find the Distinct Difference Array/Solution.cpp
class Solution {
public:
    vector<int> distinctDifferenceArray(vector<int>& nums) {
        int n = nums.size(), p = 0, s = 0;
        vector<int> prefix(51, 0), suffix(51, 0), res(n, 0);
        for (int i = 0; i < n; i++) {
            suffix[nums[i]]++;
            if (suffix[nums[i]] == 1) s++;
        }
        for (int i = 0; i < n; i++) {
            suffix[nums[i]]--;
            prefix[nums[i]]++;
            if (suffix[nums[i]] == 0) s--;
            if (prefix[nums[i]] == 1) p++;
            res[i] = p - s;
        }
        return res;
    }
};